### DISCUSSION TITLE
Single or. Multiple Models for Machine Condition Classification using Vibration Signals

### DISCUSSION THREAD

**Discussion by georgemossad** [timeseriesAI/tsai#684] [Q&A]
Greetings everyone,

I am currently working on a problem of classifying the condition of machines based on their time series vibration signals. The data I have varies in shape because each machine, such as pumps, turbines, etc., has a different number of vibration signals measured for it, leading to a different number of variables.

My question is whether I can create a single model that can handle all machines or create separate models for each machine category. Additionally, considering that pump machines comprise most of the machines in the dataset, would it be appropriate to create a single model for all machines?

Thank you for your help!

**Comment by MOREDataset** [comment]
Hi @georgemossad,

I have worked on the fault classification and detection using vibration signals. Yes, using this library you can develop time series classification models that would be for multi-variate classification. Meaning, multiple inputs or variables (of the same length). You can investigate the performance of a "machine-agnostic" classification model and a "machine-specific" classification model. Both are possible using this amazing library!

Much luck!

**Reply by georgemossad** [reply]
that's great,
so does that means I can make a model with inputs of a varied number of variables?
and also can you please tell me about the problem you were working with like what model in the library worked best for you and whether would it be better to make a single multi-model or "machine-specific" classification model?

Thank you for your help!

**Reply by MOREDataset** [reply]
Hi @georgemossad, yes you can use multiple variables of the same lengths as input to any model you chose. I would advise you to investigate multiple models from the library and see which one works best for your data. 

I worked on that problem back in 2016. I only used conventional machine learning using MATLAB. We preprocessed the data first using wavelet transform then statistical features before feeding them to different shallow models. I think a single "machine-agnostic" seems much more interesting, but only doing experimental investigations can tell you which is better. 
Hope this helps.

**Reply by georgemossad** [reply]
that's a great help @MOREDataset and nice work you have done.
do you have any idea where to begin to investigate in the library to preprocess my data if it has a varied number of variables?
many Thanks for your help!

**Reply by georgemossad** [reply]
i saw about dimensionality reduction techniques such as (PCA) or (SVD) to reduce the dimensionality of your multi-variate time series data and ensure that each time series has the same number of variables.
so is that supported to do in the library as that would be great.

**Reply by MOREDataset** [reply]
@georgemossad I would not use PCA or any other dimensionality reduction technique as you are using time series and deep learning. I would say start small with the data you have and a model of your choosing, LSTM for example, and go from there.

**Reply by georgemossad** [reply]
thanks for that.
but I still don't know how to make a model "machine-agnostic" that can handle input that has a different number of variables for each machines category.
